iroh-bytes/src/downloader.rs

```diff
@@ -1084,88 +1053,142 @@ impl<G: Getter<Connection = D::Connection>, D: Dialer> Service<G, D> {
     }
 }
 
-/// Map of potential providers for a hash.
-#[derive(Default, Debug)]
-pub struct ProviderMap {
-    /// Candidates to download a hash.
-    candidates: HashMap<Hash, HashMap<NodeId, Role>>,
-    /// Ordered list of provider hashes per node.
-    ///
-    /// I.e. blobs we assume the node can provide.
-    provider_hashes_by_node: HashMap<NodeId, VecDeque<Hash>>,
-}
-
-struct ProviderIter<'a> {
-    inner: Option<std::collections::hash_map::Iter<'a, NodeId, Role>>,
+#[derive(Debug)]
+enum NextStep {
+    StartTransfer(NodeId),
+    Dial(NodeId),
+    DialAfterIdleDisconnect(NodeId, delay_queue::Key),
```

Comment:
`DialQueuedDisconnect`?
open to other names. "After idle disconnect" sounds to me like we need to wait for the idle timeout, disconnect, and then dial. Which ofc is not what we want